src/testRunner/unittests/tsserverProjectSystem.ts

```diff
@@ -8830,4 +8841,83 @@ export const x = 10;`
             assert.equal(moduleInfo.cacheSourceFile.sourceFile.text, updatedModuleContent);
         });
     });
+
+    function makeSampleProjects() {
+        const aTs: File = {
+            path: "/a/a.ts",
+            content: "export function a() {}",
+        };
+        const aTsconfig: File = {
+            path: "/a/tsconfig.json",
+            content: `{
+                "compilerOptions": {
+                    "outDir": "bin",
+                    "declaration": true,
+                    "declarationMap": true,
+                    "composite": true,
+                }
+            }`,
+        };
+
+        const bTs: File = {
+            path: "/b/b.ts",
+            content: 'import { a } from "../a/a";\na();',
+        };
+        const bTsconfig: File = {
+            path: "/b/tsconfig.json",
+            content: `{
+                "compilerOptions": {
+                    "outDir": "bin",
+                },
+                "references": [
+                    { "path": "../a" }
+                ]
+            }`,
+        };
+
+        const host = createServerHost([aTs, aTsconfig, bTs, bTsconfig]);
+        const session = createSession(host);
+
+        writeDeclarationFiles(aTs, host, session, [
+            // Need to mangle the sourceMappingURL part or it breaks the build
+            { name: "/a/bin/a.d.ts.map", text: '{"version":3,"file":"a.d.ts","sourceRoot":"","sources":["../a.ts"],"names":[],"mappings":"AAAA,kCAAsB"}' },

```

Comment:
This sourcemap maps position `0` in the `.d.ts` to `0` in the `.ts` and `34` in the `.d.ts` to `22` in the `.ts`. The start of the declaration is mapped to the start, and the end is mapped to the end, nothing more. We try to map position `24`, which is squarely between those, so return the closest available mapping - `22`, the end mapping. `toFileSpan` in the LS then maps that into a 1-based line & column system. The end position is out of bounds, because, ofc, there's no 1-character identifier at the end of the declaration.